libcloud/compute/drivers/ec2.py

```diff
@@ -5659,7 +5659,7 @@ def __init__(self, key, secret=None, secure=True, host=None, port=None,
         self.region_name = region
         self.api_name = details['api_name']
         self.country = details['country']
-        self.signature_version = details.pop('signature_version',
+        self.signature_version = details.get('signature_version',
```

Comment:
I think this won't work because the parent class constructor method doesn't take `signature_version` argument. That's also why `pop` is used in the first place, but I could also be wrong.